ecommerce-ui/src/components/rentalForm/RentalForm.jsx

```diff
@@ -0,0 +1,172 @@
+import React, { useState } from "react";
+import DisplayCard from "../../UI/DisplayCard";
+import FormInputLine01 from "./FormInputLine01";
+import FormInputLine02 from "./FormInputLine02";
+import FormInputLine03 from "./FormInputLine03";
+import FormInputLine04 from "./FormInputLine04";
+import FormInputLine05 from "./FormInputLine05";
+import FormInputLine06 from "./FormInputLine06";
+import { Button, Grid, Typography } from "@mui/material";
+import PropTypes from "prop-types";
+
+const defaultFormValues = {
+  title: "",
+  houseType: "",
+  image: "",
+  imageAlt: "",
+  city: "",
+  country: "",
+  cost: "",
+  description: "",
+  name: "",
+  isSuperhost: false,
+};
+
+const defaultErrorState = {
+  title: [false, "Please enter a title"],
+  houseType: [false, "Please enter a rental type"],
+  image: [false, "Please enter a valid image address"],
+  imageAlt: [false, "Please enter a valid image description"],
+  city: [false, "Please enter a city"],
+  country: [false, "Please enter a country"],
+  cost: [false, "Please enter a valid amount"],
+  name: [false, "Please enter a host name"],
+  // Optional user inputs
+  description: false,
+  isSuperhost: false,
+};
+
+export default function RentalForm({ displayForm, userAddedRental }) {
+  const [formValues, setFormValues] = useState(defaultFormValues);
+  const [errorFlags, setErrorFlags] = useState(defaultErrorState);
+
+  const inputChangeHandler = (event) => {
+    const { name, value } = event.target;
+    setFormValues((currentState) => {
+      const updateState = { ...currentState };
+      if (name === "cost") {
+        updateState[name] = +value;
+      } else updateState[name] = value;
+      return updateState;
+    });
+  };
+
+  const submitHandler = (event) => {
+    event.preventDefault();
+
+    // For regex input tests
+    const emptyInput = /^\s*$/;
+
+    // Init values
+    const errorCheck = {
+      title: false,
+      houseType: false,
+      image: false,
+      imageAlt: false,
+      city: false,
+      country: false,
+      cost: false,
+      name: false,
+    };
+
+    // Complete error checks on form inputs
+    errorCheck.title = emptyInput.test(formValues.title) ? true : false;
```

Comment:
since you're doing some advance stuff here, definitely look into a library called Formik which will make handling form errors a lot easier.

https://formik.org/

Reply:
I'll definitely check this out. Thanks!